Sources/Support/StoreKitMessageType.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  StoreKitMessageType.swift
+//
+//  Created by Antonio Rico Diez on 27/9/23.
+
+import StoreKit
+
+@available(iOS 16.4, *)
+@available(macOS, unavailable)
+@available(watchOS, unavailable)
+@available(tvOS, unavailable)
+/// Types of messages available in StoreKit
+@objc public enum StoreKitMessageType: Int, CaseIterable {
```

Comment:
we're making these very StoreKit-dependent, what do we gain by abstracting them? 
I'd either: 
- make all of these "StoreMessage" such that if we have a new backing store (cough, DMA, cough) they still work 
- just vend the StoreKit types directly

I guess they're done this way for testing purposes?